FirebaseAdmin/FirebaseAdmin/Auth/TenantManager.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Google.Apis.Util;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// The tenant aware <see cref="FirebaseAuth"/> class. You can
+    /// get an instance of this class via <see cref="TenantManager.AuthForTenant(string)">FirebaseAuth.DefaultInstance.TenantManager.AuthForTenant(tenantId)</see>.
+    /// </summary>
+    public sealed class TenantManager : IFirebaseService
```

Comment:
This should not implement IFirebaseService. That should only be implemented by top-level services like `FirebaseAuth`. Just implement IDisposable here, and call it from `FirebaseAuth.Delete()`.